### ISSUE TITLE
Icinga Agent for Windows: Please add an uninstall option to decide if I want to keep the config under c:\programdata\icinga2

### ISSUE THREAD

**Issue by K0nne** [Icinga/icinga2#8038]
## Is your feature request related to a problem? Please describe.

After the uninstallation of the Icinga agent 2.11.3 it's config still exists under c:\programdata\icinga2.

## A clear and concise description of what the problem is. Ex. I'm always using this feature but am missing [...]

The content of c:\programdata is not removed, when I uninstall the icinga agent. I want a clean uninstallation.

## Describe the solution you'd like

Add an option (gui) and an argument (commandline).

## A clear and concise description of what you want to happen.

I want to have an option to decide if I want to keep the config of the icinga agent.

## Describe alternatives you've considered

Manual/scripted removal of c:\programdata\icinga2.

**Comment by Al2Klimov** [comment]
@LordHepipud What do you think a user expects to happen per default? That the config, certs and logs are kept or removed?

**Comment by LordHepipud** [comment]
We should be very careful with this change. The Icinga for Windows solution and the old Icinga 2 PowerShell Module are both requiring to uninstall and install the Agent during upgrades. If we remove this directory by default you will break many settings using this solution.

The Icinga for Windows solution is providing a argument on the uninstaller to remove this directory if the users wishes to do so.

In my opinion the best option would be to have a check box during the uninstallation prompt the user has actively to check (unchecked by default) to remove the entire ProgramData folder.

Like I said, I would not recommend removing the directory by default in any way, as it might be intentional to uninstall the Agent due to binary/service problems and I doubt users want to delete the certificates in this situation as well.

To summarize it:

* Checkbox to flush the entire ProgramData\icinga2 folder
* Print a note there to flush all certificates, config and log files as well
* The user has actively to check the checkbox for this process

I'm open to the CLI argument for uninstalling it, as it wouldn't affect me. Both PowerShell solutions ship with this feature anyway already so that one is up to you :D

**Comment by Al2Klimov** [comment]
To me the MSI stuff seems not to allow a such feature.

![Bildschirmfoto 2020-11-02 um 12 47 23](https://user-images.githubusercontent.com/6053266/97866923-23f57000-1d0d-11eb-8361-44f0c1adeeae.png)
![Bildschirmfoto 2020-11-02 um 12 47 30](https://user-images.githubusercontent.com/6053266/97866911-1f30bc00-1d0d-11eb-9ed9-25437f123adc.png)
![Bildschirmfoto 2020-11-02 um 12 47 48](https://user-images.githubusercontent.com/6053266/97866882-150ebd80-1d0d-11eb-83d6-79fc320d3e2b.png)

![Bildschirmfoto 2020-11-02 um 13 05 01](https://user-images.githubusercontent.com/6053266/97867048-61f29400-1d0d-11eb-8408-9d715885c08a.png)
![Bildschirmfoto 2020-11-02 um 13 05 13](https://user-images.githubusercontent.com/6053266/97867064-6b7bfc00-1d0d-11eb-8d06-1ff4c486945c.png)

@LordHepipud Correct me if I'm wrong.

@K0nne Consider using Icinga for Windows:

> The Icinga for Windows solution is providing a argument on the uninstaller to remove this directory if the users wishes to do so.

**Comment by LordHepipud** [comment]
I think this relates on how an MSI installer is build. As far as I know you would be able to customize some features and actions, but I would need to check this more detailed.

**Comment by Al2Klimov** [comment]
As I've got your attention already: Do you consider the following a reasonable alternative for users?

> @K0nne Consider using Icinga for Windows:
> 
> > The Icinga for Windows solution is providing a argument on the uninstaller to remove this directory if the users wishes to do so.

**Comment by K0nne** [comment]
Awesome! I wasn't aware of this. Is this already in the docs? I searched, but found nothing.

**Comment by Al2Klimov** [comment]
1. So you'll switch to Icinga for Windows and don't need this feature anymore, right?
2. Docs (to do): #8141

**Comment by K0nne** [comment]
1. Yes, we are about to switch to Icinga for windows.
2. Thanks! 

I close this.
